backend/fms_core/services/dataset.py

```diff
@@ -0,0 +1,195 @@
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+
+from datetime import datetime, date
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from fms_core.models.dataset_file import DatasetFile
+from fms_core.models.dataset import Dataset
+
+def convertToModelDate(value: Union[datetime, date, str, None]) -> Union[datetime, None]:
+    if isinstance(value, date):
+        return datetime.combine(value, datetime.min.time())
+    elif isinstance(value, datetime):
+        return value
+    elif isinstance(value, str):
+        return datetime.fromisoformat(value[:10])
+    elif value is None:
+        return value
+    else:
+        raise ValueError(f"Invalid date: {value}")
+
+def create_dataset(project_name: str, run_name: str, lane: str, files: List[Dict[str, Any]] = []) -> Tuple[Union[Dataset, None], List[str], List[str]]:
+    dataset = None
+
+    errors = []
+    warnings = []
+
+    try:
+        dataset = Dataset.objects.create(
+            project_name=project_name,
+            run_name=run_name,
+            lane=lane,
+        )
+    except ValidationError as e:
+        errors.append(';'.join(e.messages))
+    
+    
+    dataset_files = []
+
+    for file in files:
+        if not errors:
+            dataset_file, newerrors, newwarnings = create_dataset_file(dataset, file['file_path'], file['sample_name'])
+            errors.extend(newerrors)
+            warnings.extend(newwarnings)
+
+            if dataset_file:
+                dataset_files.append(dataset_file)
+
+    if errors:
+        Dataset.objects.filter(id__in=dataset if dataset else []).delete()
+        DatasetFile.objects.filter(id__in=dataset_files).delete()
+
+        dataset = None
+        dataset_files = []
+
+    return (dataset and Dataset.objects.get(pk=dataset.id), errors, warnings)
+
+def create_dataset_file(dataset: Dataset, file_path: str, sample_name: str, completion_date: Optional[datetime] = None, validation_date: Optional[datetime] = None) -> Tuple[Union[DatasetFile, None], List[str], List[str]]:
+    dataset_file = None
+    errors = []
+    warnings = []
+
+    try:
+        dataset_file = DatasetFile.objects.create(
+            dataset=dataset,
+            file_path=file_path,
+            sample_name=sample_name,
+            completion_date=completion_date,
+            validation_date=validation_date,
+        )
+    except ValidationError as e:
+        errors.append(';'.join(e.messages))
+
+    return (dataset_file, errors, warnings)
+
+def create_from_run_processing(run_processing_metrics: Dict[str, Any], completion_date: str, validation_date: str) -> Tuple[List[Dataset], List[DatasetFile], List[str], List[str]]:
```

Comment:
I would this out of the services. Services are a layer we want to keep a bit insulated from the incoming structure of data in order to be able to use and reuse it from different sources. This would belong to somewhere similar to the template importers in a more generic fashion.